src/components/Header/UserPreferences/Automation/Decrypt.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ReactElement, useEffect, useRef, useState } from 'react'
+import styles from './Decrypt.module.css'
+import Button from '../../../@shared/atoms/Button'
+import { toast } from 'react-toastify'
+import { useAutomation } from '../../../../@context/Automation/AutomationProvider'
+import Loader from '../../../@shared/atoms/Loader'
+import InputElement from '../../../@shared/FormInput/InputElement'
+
+export default function Decrypt(): ReactElement {
+  const {
+    isLoading,
+    decryptPercentage,
+    decryptAutomationWallet,
+    setIsAutomationEnabled
+  } = useAutomation()
+
+  const decrpytToastRef = useRef(null)
+  const passwordInputRef = useRef(null)
+
+  useEffect(() => {
+    toast.update(decrpytToastRef.current, { progress: decryptPercentage })
+  }, [decrpytToastRef, decryptPercentage])
+
+  return (
+    <div className={styles.wrapper}>
+      {isLoading ? (
+        <Loader message="Decrypting..." />
+      ) : (
+        <>
+          <strong className={styles.warning}>The wallet is locked!</strong>
+          <form
+            onSubmit={async (e) => {
+              e.preventDefault()
+              decrpytToastRef.current = toast.info(`Decrypting Wallet...`)
+              await decryptAutomationWallet(passwordInputRef.current.value)
+              setIsAutomationEnabled(true)
+              toast.done(decrpytToastRef.current)
```

Comment:
question: What happens if the decryptAutomationWallet function fails? It seems the toast will still indicate success with toast.done.

Reply:
suggestion: Add error handling for the decryption process. If the decryptAutomationWallet function fails, provide feedback to the user, possibly using a toast error message. Or we just rely on the handling by the automation?

Reply:
The `toast.done` only manages the state of the "progress" toast message. With done we will close this message in all states (failure & success)

The error or success states are handled separately in the [`decryptAutomationWallet` callback within the `AutomationProvider`](https://github.com/deltaDAO/mvg-portal/blob/bd1091bbd03677a39de9ed6a053cc15e0612acd7/src/%40context/Automation/AutomationProvider.tsx#L188-L195)